src/robo_jogador.py

```
@@ -21,77 +45,173 @@ def sense_act(self, stdscr):
+                                with self.locks['robots_mutex']:
+                                    self.pos = target
+                                    self.robots_info[self.id]['pos'] = self.pos
+                                    self.E -= 1 # Consome energia ao mover
+                                    self.robots_info[self.id]['E'] = self.E
+                                self.logger.info(f"Robo {self.id} moveu de {old_pos} para {self.pos}. Energia: {self.E}")
+                                
+                        elif cell_content == '⚡': # Célula com bateria, tenta coletar
+                            battery_lock = self.locks['battery_mutexes'].get(target)
+                            if battery_lock:
+                                with self.locks['grid_mutex']: # grid_mutex para acessar e modificar o grid
+                                    with battery_lock: # battery_mutex_k para a bateria específica
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)